Bundle/Controller/OpenApi/CategoriesController.php

```diff
@@ -0,0 +1,341 @@
+ */
+class CategoriesController extends Controller
+{
+    /* @var ErrorCollectValidator */
+    protected $validator;
+    /* @var TokenStorage */
+    protected $tokenStorage;
+    /* @var integer */
+    protected $subscribersLimit;
+    /* @var Categories */
+    private $model;
+
+    public function __construct(Categories $model, ErrorCollectValidator $validator, TokenStorage $tokenStorage, $subscribersLimit)
+    {
+        $this->model = $model;
+        $this->validator = $validator;
+        $this->tokenStorage = $tokenStorage;
+        $this->subscribersLimit = $subscribersLimit;
+    }
+
+    /**
+     * @Route("/categories")
+     * @Method("GET")
+     * @View(serializerGroups={"user"})
+     */
+    public function getCategoriesAction(ParamFetcher $paramFetcher)
+    {
+        $this->validator->validateAndThrow($paramFetcher);
+
+        $s = $this->model->select();
+        $s->whereEquals('newsletter_component_id', $this->tokenStorage->getToken()->getUser()->getNewsletterComponent()->dbId);
+
+        return $this->model->getRows($s);
+    }
+
+    /**
+     * @Route("/categories")
+     * @RequestParam(name="category", requirements=".+", strict=true, nullable=false)
+     * @Method("POST")
+     * @View(serializerGroups={"user"}, statusCode=201)
+     */
+    public function postCategoryAction(ParamFetcher $paramFetcher, Request $request)
+    {
+        $this->validator->validateAndThrow($paramFetcher);
+
+        $row = $this->model->createRow(array(
+            'newsletter_component_id' => $this->tokenStorage->getToken()->getUser()->getNewsletterComponent()->dbId,
+        ));
+        $this->updateRow($row, $paramFetcher);
+        $row->save();
+
+        return $row;
+    }
+
+    protected function updateRow(\Kwf_Model_Row_Abstract $row, ParamFetcher $paramFetcher)
+    {
+        $row->category = $paramFetcher->get('category');
+    }
+
+    /**
+     * @Route("/categories/{id}", requirements={"id"="[1-9]{1}\d*"})
+     * @RequestParam(name="category", requirements=".+", strict=true, nullable=false)
+     * @Method("PUT")
+     * @View(serializerGroups={"user"})
+     */
+    public function putCategoryAction($id, ParamFetcher $paramFetcher, Request $request)
+    {
+        $this->validator->validateAndThrow($paramFetcher);
+
+        $newsletterComponent = $this->tokenStorage->getToken()->getUser()->getNewsletterComponent();
+
+        $s = $this->model->select();
+        $s->whereEquals('id', $id);
+        $s->whereEquals('newsletter_component_id', $newsletterComponent->dbId);
+        $row = $this->model->getRow($s);
+
+        if (!$row) {
+            throw new NotFoundHttpException($newsletterComponent->trlKwf('Category not found'));
+        }
+
+        $this->updateRow($row, $paramFetcher);
+
+        if ($row->isDirty()) {
+            $row->save();
+        }
+
+        return null;
+    }
+
+    /**
+     * @Route("/categories/{id}/subscribers", requirements={"id"="[1-9]{1}\d*"})
```

Comment:
return only ids (can be 10000 - test performance for that case)

Reply:
Changed to ids.  With 12000+ subscribers it takes 7-9 seconds to return the ids (on the test vm)